src/main/java/org/dsa/iot/alarm/AlarmAlgorithm.java

```diff
@@ -0,0 +1,362 @@
+/* THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH REGARD
+ * TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS. IN
+ * NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL
+ * DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER
+ * IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+ * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+package org.dsa.iot.alarm;
+
+import org.dsa.iot.dslink.node.*;
+import org.dsa.iot.dslink.node.actions.*;
+import org.dsa.iot.dslink.node.value.*;
+import org.dsa.iot.dslink.util.*;
+import org.dsa.iot.dslink.util.json.*;
+import java.util.concurrent.*;
+
+/**
+ * Alarm algorithms evaluate the state of AlarmWatch objects, and generate alarms for
+ * each when the conditions of the algorithm are met.
+ * <p/>
+ * Subclasses should override the following methods:
+ * <ul>
+ * <li>isAlarm(AlarmWatch).</li>
+ * <li>getAlarmMessage(AlarmWatch).</li>
+ * </ul>
+ *
+ * @author Aaron Hansen
+ */
+public abstract class AlarmAlgorithm extends AbstractAlarmObject implements Runnable {
+
+    ///////////////////////////////////////////////////////////////////////////
+    // Constants
+    ///////////////////////////////////////////////////////////////////////////
+
+    private static final String ALARM_TYPE = "Alarm Type";
+    private static final String AUTO_UPDATE_INTERVAL = "Auto Update Interval";
+    private static final String TO_ALARM_INHIBIT = "To Alarm Inhibit";
+    private static final String TO_NORMAL_INHIBIT = "To Normal Inhibit";
+
+    ///////////////////////////////////////////////////////////////////////////
+    // Fields
+    ///////////////////////////////////////////////////////////////////////////
+
+    private AlarmState alarmState; //alert, fault or offnormal
```

Comment:
that comment is useless. We'll go see the `AlamState` enum if we want to know. Also, it risks being outdated if the enum changes.